tests/behavior_test.cpp

```diff
@@ -0,0 +1,151 @@
+    // Got clean water, proceed to food.
+    CHECK( maslows.tick( nullptr ) == "get_food" );
+    water_state = behavior::failure;
+    clean_water_state = behavior::running;
+    // Failed to get water, give up.
+    CHECK( maslows.tick( nullptr ) == "get_food" );
+    water_state = behavior::running;
+    CHECK( maslows.tick( nullptr ) == "get_water" );
+    cold_state = behavior::success;
+    thirsty_state = behavior::success;
+    hungry_state = behavior::running;
+    // Second need also met, third branch taken.
+    CHECK( maslows.tick( nullptr ) == "get_food" );
+    cold_state = behavior::success;
+    // Failure also causes third branch to be taken.
+    CHECK( maslows.tick( nullptr ) == "get_food" );
+    thirsty_state = behavior::success;
+    // Failure in second branch too.
+    CHECK( maslows.tick( nullptr ) == "get_food" );
+    thirsty_state = behavior::running;
+    cold_state = behavior::running;
+    // First need appears again and becomes highest priority again.
+    CHECK( maslows.tick( nullptr ) == "wear" );
+    clothes_state = behavior::failure;
+    // First alternative failed, attempting second.
+    CHECK( maslows.tick( nullptr ) == "fire" );
+    fire_state = behavior::failure;
+    // Both alternatives failed, check other needs.
+    CHECK( maslows.tick( nullptr ) == "get_water" );
+    clothes_state = behavior::success;
+    // Either failure or success meets requirements.
+    CHECK( maslows.tick( nullptr ) == "get_water" );
+    clothes_state = behavior::failure;
+    fire_state = behavior::success;
+    // Either order does it.
+    CHECK( maslows.tick( nullptr ) == "get_water" );
+    hungry_state = behavior::running;
+    // Still thirsty, so changes to hunger are irrelevant.
+    CHECK( maslows.tick( nullptr ) == "get_water" );
+    thirsty_state = behavior::success;
+    hungry_state = behavior::success;
+    // All needs met, so no goals remain.
+    CHECK( maslows.tick( nullptr ) == "idle" );
+}
+
+// Make assertions about loaded behaviors.
+TEST_CASE( "check_npc_behavior_tree", "[behavior]" )
+{
+    behavior::tree npc_needs;
+    npc_needs.add( &string_id<behavior::node_t>( "npc_needs" ).obj() );
+    npc test_npc;
+    test_npc.normalize();
+    test_npc.setpos( { 50, 50, 0 } );
+    behavior::character_oracle_t oracle( &test_npc );
+    CHECK( npc_needs.tick( &oracle ) == "idle" );
+    SECTION( "Freezing" ) {
+        g->temperature = 0;
+        test_npc.update_bodytemp();
+        CHECK( npc_needs.tick( &oracle ) == "idle" );
+        item &sweater = test_npc.i_add( item( itype_id( "sweater" ) ) );
+        CHECK( npc_needs.tick( &oracle ) == "wear_warmer_clothes" );
+        item sweater_copy = test_npc.i_rem( &sweater );
+        test_npc.wear_item( sweater_copy );
+        CHECK( npc_needs.tick( &oracle ) == "idle" );
+        test_npc.i_add( item( itype_id( "lighter" ) ) );
+        test_npc.i_add( item( itype_id( "2x4" ) ) );
+        CHECK( npc_needs.tick( &oracle ) == "start_fire" );
+    }
```

Comment:
Add test cases for eating food and drinking water to satisfy hunger and thirst?

Reply:
I backed myself into a corner here, I gave up on writing a predicate to evaluate searching for food, so the food branch will always return failure.

With some distance from that initial decision though, I can just write a predicate to check inventory for food and water and have at least some basic functionality here.